src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -286,20 +288,21 @@ trait ClusterComponent extends LeoComponent {
       )
     }
 
-    private def unmarshalClustersWithLabels(clusterLabels: Seq[(ClusterRecord, Option[LabelRecord])]): Seq[Cluster] = {
+    private def unmarshalClustersWithLabels(clusterLabels: Seq[(ClusterRecord, Option[LabelRecord], Option[ClusterErrorRecord])]): Seq[Cluster] = {
       // Call foldMap to aggregate a Seq[(ClusterRecord, LabelRecord)] returned by the query to a Map[ClusterRecord, Map[labelKey, labelValue]].
-      val clusterLabelMap: Map[ClusterRecord, LabelMap] = clusterLabels.toList.foldMap { case (clusterRecord, labelRecordOpt) =>
+      val clusterLabelMap = clusterLabels.toList.foldMap { case (clusterRecord, labelRecordOpt, errorOpt) =>
         val labelMap = labelRecordOpt.map(labelRecordOpt => labelRecordOpt.key -> labelRecordOpt.value).toMap
-        Map(clusterRecord -> labelMap)
+        val errorList = errorOpt.toList.groupBy(_.id).map(_._2.head).map(e => ClusterError(e.errorMessage, e.errorCode, e.timestamp.toInstant)).toList
+        //println(errorList)
+        Map(clusterRecord -> (labelMap, errorList))
       }
-
       // Unmarshal each (ClusterRecord, Map[labelKey, labelValue]) to a Cluster object
-      clusterLabelMap.map { case (clusterRec, labelMap) =>
-        unmarshalCluster(clusterRec, labelMap)
+      clusterLabelMap.map { case (clusterRec,(labelMap, error)) =>
+        unmarshalCluster(clusterRec, labelMap, error.groupBy(_.timestamp).map(_._2.head).toList)
       }.toSeq
     }
 
-    private def unmarshalCluster(clusterRecord: ClusterRecord, labels: LabelMap): Cluster = {
+    private def unmarshalCluster(clusterRecord: ClusterRecord, labels: LabelMap, error:List[ClusterError]): Cluster = {
```

Comment:
It would be nice if this method took `ClusterErrorRecord` instead of `ClusterError`, and handled the unmarshalling from records to model objects.